apps/examples/src/mcp-discord-agent/index.ts

```diff
@@ -0,0 +1,113 @@
+import {
+	createSamplingHandler,
+	LlmAgent,
+	Runner,
+	createDatabaseSessionService,
+	InMemoryMemoryService,
+	type SamplingHandler,
+	McpDiscord,
+} from "@iqai/adk";
+import * as fs from "node:fs";
+import * as path from "node:path";
+import { env } from "node:process";
+
+const APP_NAME = "McpDiscordAgent";
+const USER_ID_PREFIX = "discord-user-";
+let runner: Runner;
+let samplingHandler: SamplingHandler;
+
+async function main() {
+	// Create a sampling handler for MCP
+	samplingHandler = createSamplingHandler(async (llmRequest) => {
+		const lastMsg = llmRequest.contents[llmRequest.contents.length - 1];
+		const content = lastMsg.parts[0].text!;
+		const userIdMatch = content.match(/user_id:\s*(\d+)/);
+		const userId = userIdMatch
+			? `${USER_ID_PREFIX}${userIdMatch[1]}`
+			: `${USER_ID_PREFIX}unknown`;
+		console.log(userId);
+		let response = "";
+		for await (const event of runner.runAsync({
+			userId,
+			sessionId: await getOrCreateUserSession(userId),
+			newMessage: lastMsg,
+		})) {
+			if (
+				event.content?.parts &&
+				event.author === "discord_agent" &&
+				!event.partial
+			) {
+				response += event.content.parts.map((p) => p.text).join("");
+			}
+		}
+		return {
+			content: { role: "model", parts: [{ text: response }] },
+		};
+	});
+	await initializeRunner();
+	process.stdin.resume();
+}
```

Comment:
we need make this much more simpler i believe we need a new method in runner that would give us the final response instead of stream. need to let the createSamplingHandler take string as output along with LlmResponse